### DISCUSSION TITLE
How to set Authorization for SendHttpRequest activity?

### DISCUSSION THREAD

**Discussion by jayachandra21** [elsa-workflows/elsa-core#1658] [Q&A]
Hi @sfmskywalker,

How to set Authorization permissions let say windows authorization for **SendHttpRequest** activity?

**Comment by sfmskywalker** [comment]
The way to do it is the same way you would do it when using `HttpClient`: by setting Headers. For example, when using Basic Auth, you would set the `Authorization` header with a value of `Basic <credentials>`.

Depending on your authentication scheme, you need to provide a specific form of credentials. This is the same whether you use `HttpClient` or the `SendHttpRequest` activity (which uses `HttpClient` under the hood).

More information about various authentication schemes can be found here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#authentication_schemes

**Reply by landonzeng** [reply]
So how do I set the Authorization value of SendHttpRequest on the dashboard to be the Authorization value passed in by HttpEndpoint

**Reply by sfmskywalker** [reply]
Good question. I haven't tried, but what you could do is configure your authentication middleware to [save the incoming token](https://www.jerriepelser.com/blog/aspnetcore-jwt-saving-bearer-token-as-claim/).

Then you need to access this token from the current HTTP context.

However, there's currently no JS function or Liquid tag for this, but [it's easy to add](https://elsa-workflows.github.io/elsa-core/docs/next/extensibility/extensibility-javascript).

When you have a JS function (e.g. named `getAccessToken`) that returns the stored token, you use it in the `SendHttpRequest` by setting an `Authorization` header and construct the value using your token, e.g.:

```javascript
`Bearer ${getAccessToken()}`
```

If you want to create a PR that adds this JavaScript (and/or Liquid) then that would be even better 😄
